spec/models/course/condition/achievement_spec.rb

```diff
@@ -86,6 +86,13 @@
       end
     end
 
+    describe '#dependent_object' do
+      it 'returns the correct dependent achievement object' do
+        subject.achievement = create(:course_achievement, course: course)
```

Comment:
actually this should be part of the condition's attributes. can you create the achievement condition together with the achievement, using the Factory Girl factory?